src/Nixfmt/Util.hs

```diff
@@ -0,0 +1,52 @@
+module Nixfmt.Util
+    ( manyList
+    , someList
+    , manyP
+    , someP
+    , manyText
+    , someText
+    , commonPrefix
+    , dropCommonIndentation
+    ) where
+
+import           Data.Char
+import           Data.Maybe
+import           Data.Text       as Text hiding (map, takeWhile)
+import           Data.Void
+import           Text.Megaparsec hiding (empty)
+
+type Parser = Parsec Void Text
+
+someP :: (Char -> Bool) -> Parser Text
+someP = takeWhile1P Nothing
+
+manyP :: (Char -> Bool) -> Parser Text
+manyP = takeWhileP Nothing
```

Comment:
These names are a little misleading, as the `some` and `many` functions usually do something different 🤔.

Reply:
To be honest, I feel these names are very accurate. They are variants of `some` and `many` with some additional convenience. In fact, I feel like the `takeWhileP` names stand out for doing something very similar, but being named completely different.